### PR TITLE
[test] Add ert roofline check (gpu)

### PR DESCRIPTION
**Pull request by jgphpc**
Will fix UES-213 (gpu)

### CODE REVIEW

File: cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                sn.assert_reference(sn.extractsingle(
+                    r'(?P<GFLOPs>\d+.\d+)\sGFLOPs EMP', self.roofline_rpt,
+                    'GFLOPs', float), GFLOPs, -0.1, 0.5),
+                # check L1 bandwidth:
+                sn.assert_reference(sn.extractsingle(
+                    r'(?P<L1bw>\d+.\d+)\sL1 EMP', self.roofline_rpt,
+                    'L1bw', float), L1bw, -0.1, 0.3),
+                # check DRAM bandwidth:
+                sn.assert_reference(sn.extractsingle(
+                    r'(?P<DRAMbw>\d+.\d+) DRAM EMPv', self.roofline_rpt,
```
Comment (jgphpc) [inline cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py]: EMP`v` is a typo but fine

**Comment by jgphpc** [comment]
> Batch job submission failed: Socket timed out on send/recv operation

File: cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        # A single input file is required for nvcc to work:
+        self.build_system = 'SingleSource'
+        self.prebuild_cmd = [
+            'cat kernel1.c driver1.c | sed "s-^#if ERT-#ifdef ERT-g" > '
+            '_gpu.cu']
+        self.sourcepath = '_gpu.cu'
+        self.executable = 'ert.exe'
+        self.build_system.cppflags = [
+            # ERT_FLOPS = -DERT_FLOP !
+            '-DERT_FLOP=%s' % str(flop),
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py]: No need for `str(...)` here.

File: cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            '-DERT_GPU',
+            '-DERT_TRIALS_MIN=1',
+            '-DERT_WORKING_SET_MIN=128',
+            # '-x cu' explicitly sets the language (cuda) for the src files.
+        ]
+        self.build_system.ldflags = ['-O3']
+        self.maintainers = ['JG']
+        self.tags = {'scs'}
+        gpu_blocks = gpudims[0]
+        gpu_threads = gpudims[1]
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py]: Rewrite as:

```suggestion
        gpu_blocks, gpu_threads = gpudims
```

File: cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+class ErtP100Test(rfm.RegressionTest):
+    """
+    The Empirical Roofline Tool, ERT, empirically generates roofline data:
+    https://bitbucket.org/berkeleylab/cs-roofline-toolkit/
+
+    This test checks the ERT tool with NVIDIA Tesla P100-PCIE-16GB:
+    Device 0: "Tesla P100-PCIE-16GB"
+      CUDA Driver Version / Runtime Version     10.1 / 10.0
+      CUDA Capability Major/Minor version number:    6.0
+      (56) Multiprocessors, ( 64) CUDA Cores/MP:     3584 CUDA Cores
```
[the remark this answers is not in the document]
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/berkeley-ert-nvprof.py]: https://github.com/eth-cscs/reframe/pull/775/commits/beb7d7a1514c413fc1d3d0b7f113ed659fe1d7a2

**Comment by vkarak** [comment]
@teojgo Can you approve this PR?

**Comment by teojgo** [comment]
@vkarak there is one particular case that fails in the sanity step

**Comment by teojgo** [comment]
@vkarak you can merge it now

**Comment by vkarak** [comment]
@jenkins-cscs retry none

**Pull request merged**